pkg/controller/gpudevice/gpu_controller.go

```diff
@@ -267,3 +278,86 @@ func isNotDriverPod(pod *corev1.Pod) bool {
 
 	return true
 }
+
+// setupMigconfiguration will check if GPU suports MIG instances
+// and accordingly creates a MIGConfigurationObject if one is not already present
+func (h *Handler) setupMigConfiguration(gpu *v1beta1.SRIOVGPUDevice) error {
+	// check if GPU supports MIG instances
+	ok, err := gpuhelper.IsMigConfigurationNeeded(h.executor, gpu.Spec.Address)
+	if err != nil {
+		return fmt.Errorf("error checking if GPU supports MIG instances: %w", err)
+	}
+
+	// if MIGConfiguration is not enabled and
+	if !ok {
+		return nil
+	}
+
+	// enable MIGMode on device, this is idempotent so can be run multiple times
+	if err := gpuhelper.EnableMIGMode(h.executor, gpu.Spec.Address); err != nil {
+		return fmt.Errorf("error enable MIG Mode: %w", err)
+	}
+
+	// MIGConfiguration object has same name as GPU Name so we can use that to lookup existing device
+	obj, err := h.migConfigurationCache.Get(gpu.Name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("error looking up MIG Configuration: %w", err)
+	}
+
+	if obj != nil {
+		// existing object found, nothing else is needed
+		return nil
+	}
+
+	migConfiguration, err := gpuhelper.GenerateMIGConfiguration(h.executor, gpu.Spec.Address, gpu.Name)
+	if err != nil {
+		return fmt.Errorf("error generating MIG Configuration: %w", err)
+	}
+
+	migStatus, err := gpuhelper.GenerateMIGConfigurationStatus(h.executor, gpu.Spec.Address)
+	if err != nil {
+		return fmt.Errorf("error generating MIG configuration status: %w", err)
+	}
+
+	migConfiguration.Spec.NodeName = gpu.Spec.NodeName
+	if instanceCount(migStatus) > 0 {
+		migConfiguration.Spec.Enabled = true
+	}
```

Comment:
Can we just let the MIG controller reconciles the status by setting `migConfiguration.Spec.Enabled = true` here? Then the `if mig.Spec.Enabled && instanceCount(status) == 0` condition in the MIG controller should handle the rest.

Reply:
Never mind.. I just remember now how the instanceCount is calculated.